gh-139038: Credit more people for JIT in 3.15 (#148359)
python/cpython: Doc/whatsnew/3.15.rst

optimizer. This allows the JIT compiler to avoid certain stack operations
altogether and instead operate on registers. This allows the JIT to produce
more efficient traces by avoiding reads and writes to memory.
(Contributed by Mark Shannon in :gh:`135379`.)

.. rubric:: More JIT optimizations

More `constant-propagation <https://en.wikipedia.org/wiki/Constant_folding>`__
is now performed. This means when the JIT compiler detects that certain user
code results in constants, the code can be simplified by the JIT.
(Contributed by Ken Jin and Savannah Ostrowski in :gh:`132732`.)

> :term:`reference count`\ s are avoided where safe to do so. This generally

Reviewer: ```suggestion
:term:`reference count`\ s are avoided whenever it is safe to do so. This generally
```

Author: Thank you.

> :term:`reference count`\ s are avoided where safe to do so. This generally

Reviewer 2: ```suggestion
:term:`Reference count`\ s are avoided where safe to do so. This generally
```
I think it'll still work?

reduces the cost of most operations in Python.
(Contributed by Ken Jin, Donghee Na, Zheao Li, Hai Zhu, Savannah Ostrowski,
Reiden Ong, Noam Cohen, Tomas Roun, PuQing, Cajetan Rodrigues, and Sacul in :gh:`134584`.)

> The JIT optimizer now supports significantly more operations than in 3.14.

Reviewer: ```suggestion
The JIT optimizer now supports significantly more operations than the JIT optimizer in 3.14
```

Small suggestion for clarity

> (Contributed by Kumar Aditya, Ken Jin, and Sacul :gh:`131798`)

Reviewer 2: ```suggestion
(Contributed by Kumar Aditya, Ken Jin, and Sacul in :gh:`131798`)
```

> (Contributed by Kumar Aditya, Ken Jin, and Sacul in :gh:`131798`)

Reviewer 2: ```suggestion
(Contributed by Kumar Aditya, Ken Jin, and Sacul in :gh:`131798`.)
```
Sorry, I missed it before :-(

.. rubric:: Better machine code generation

The JIT compiler's machine code generator now produces better machine code
for x86-64 and AArch64 macOS and Linux targets. In general, users should
experience lower memory usage for generated machine code and more efficient

> machine code versus 3.14.

Reviewer: ```suggestion
machine code versus the machine code generated by the 3.14 JIT compiler.
```

Same here

Author: I think this is already implied. It would also be dishonest of us to compare to something else, so I hope people think the same.
